### PR TITLE
FEAT: Add Single Product Page

### PR DESCRIPTION
**Pull request by VishalPatil18**
### 📝 Description

A single product page is one having all details of a single product. You could add a product to the cart and wishlist from here or also buy from this page.

### 🚀 New behavior

- A single product page appears when a user clicks on any card
- The user could add the product to a wishlist, cart, or directly buy it. (Buy is not yet implemented)
- The product page is dynamic for all products

### 🖼 Preview link

https://viskart-react-version-qtd4j1eg4-vishalpatil18.vercel.app/

### 💣 Is this a breaking change?
- [ ] Yes
- [x] No

### 🔬 Is the code self-reviewed?
- [x] Yes
- [ ] No

### 📋 Is the code well-formatted?
- [x] Yes
- [ ] No

### 🔗 Related to Issue 
- #18 

> NOTE: The site is not responsive yet. Will work on responsiveness at the end

### CODE REVIEW

File: src/components/SingleProductPage/SingleProductPage.jsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  );
+  const [isWishlisted, setIsWishlisted] = useState(
+    isInArrayList(product, wishlistState.wishlist)
+  );
+
+  useEffect(() => {
+    setLoader(true);
+    (async () => {
+      try {
+        const productFromResponse = await getSingleProductHandler(productID);
```
Comment (mohammed-sas) [inline src/components/SingleProductPage/SingleProductPage.jsx:43]: check the response status before updating the state , you havent checked for status in handler as well as in service

**Review by mohammed-sas — APPROVED** [review]
Code is clean and readable
Single product page functionalities are working as intended
Overall code is looking fine

**Comment by starver20** [comment]
You need not mention the price of the product on the buy and wishlist button since you have already mentioned the price on the card above.

**Comment by starver20** [comment]
Code is well written, good to go👍

File: src/components/SingleProductPage/SingleProductPage.jsx (hunk 1 above)
Comment (VishalPatil18) [reply src/components/SingleProductPage/SingleProductPage.jsx:43]: Sure, will resolve it in the next PR.

**Review by starver20 — APPROVED** [review]

**Pull request merged**